### ISSUE TITLE
add_tags() API is inconsistent

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dhamaniasad/mutagen#101]
```
All of the mutagen object types that support an 'add_tags()' function do so in 
the following way (mostly):

def add_tags(self):
    if self.tags:
        self.tags = Something()
    ... 

Whereas the MP4 add_tags function (line 670) always overwrites self.tags, 
regardless of whether there are pre-existing tags or not. This could cause 
external developers to accidentally overwrite existing tags in a file if they 
don't pay attention to this difference. 
```

Original issue reported on code.google.com by `dus...@virtualroadside.com` on 7 Dec 2011 at 4:21

**Comment by GoogleCodeExporter** [comment]
Original comment by `reiter.christoph@gmail.com` on 25 Jul 2012 at 2:07
- Changed state: **Accepted**

Attachments:
- [mp4-add-tags-check-v1.patch](https://storage.googleapis.com/google-code-attachments/mutagen/issue-101/comment-1/mp4-add-tags-check-v1.patch)

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 591f00fe3bcd.
```

Original comment by `reiter.christoph@gmail.com` on 2 Apr 2013 at 4:35
- Changed state: **Fixed**
